### DISCUSSION TITLE
Can Shell MenuItems be disabled?

### DISCUSSION THREAD

**Discussion by matthewclendening** [dotnet/maui#9630] [Q&A]
I can't figure out how to make shell menuitems render in a disabled state. If I have a menu item that binds to a command, I'd like it if that command's canexecute could disable the menuitem if it returned false. Is this an intended use case for these menuitems ?

Update:
If you have the following:
`<Shell
    x:Class="MauiAppWithMenuItem.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:MauiAppWithMenuItem"
    x:DataType="local:ShellViewModel"
    Shell.FlyoutBehavior="Flyout">

    <FlyoutItem Title="Home">
        <ShellContent ContentTemplate="{DataTemplate local:MainPage}" Route="MainPage" />
    </FlyoutItem>

    <MenuItem Text="MyCommand" Command="{Binding MyActionCommand}"  IsEnabled="False" />
</Shell>
`

This will throw System.InvalidOperationException: 'The BindableProperty "IsEnabled" is readonly.'

**Comment by melixor** [comment]
seems  this function to be impossible?

**Reply by Xyncgas** [reply]
Fork the MAUI repository and build your application on it, pull master branch occasionally and modify anything you want, if it doesn't let you modify a value just modify it so it lets you modify the value

**Comment by PureWeen** [comment]
https://github.com/dotnet/maui/issues/9492
